### PR TITLE
Templating getWithSelected() and getCheckbox(), Issue #12004

### PR DESCRIPTION
**Pull request by 007durgesh219**
Before submitting pull request, please check that every commit:
- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them

Signed-off-by: Durgesh 007durgesh219@gmail.com

### CODE REVIEW

**Comment by 007durgesh219** [comment]
This fails because of following.  I think here we need to change the tests.
```
[exec] --- Expected
     [exec] +++ Actual
     [exec] @@ @@
     [exec] -'<input type="checkbox" name="test_display_html_checkbox"
     [exec] -                    id="test_display_html_checkbox"
     [exec] -        <label              for="test_display_html_checkbox">text_label_for_checkbox</label>
     [exec] -'
     [exec] +'<input type="checkbox" name="test_display_html_checkbox" id="test_display_html_checkbox" /><label for="test_display_html_checkbox">text_label_for_checkbox</label>'
```

**Comment by 007durgesh219** [comment]
@zixtor Kindly help

**Comment by zixtor** [comment]
Indeed, the tests might be failing due of some indentation/formatting changes, and need to be updated accordingly.

**Comment by 007durgesh219** [comment]
Thanks, I will update test and PR shortly

**Comment by zixtor** [comment]
You may also choose to assert for something less than full html tags.

**Comment by 007durgesh219** [comment]
So, Can I use DOMDocument for asserting only attributes in which we are interested so that in future if we change the position of attributes it won't fail??. Or please suggest something similar to assert only attributes.

**Comment by 007durgesh219** [comment]
Or what about assertTag.  [https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.assertions.assertTag](url)

**Comment by zixtor** [comment]
Yes, we should tend to avoid recurring of such failures, asserting for attributes and their values as essential in the generated html seems fine to me. Using DOMDocument for that seems promising to me at first look, I think it would also validate the generated html? Also, assertTag doesn't seem to be a reliable choice and is deprecated, see https://github.com/sebastianbergmann/phpunit/issues/1292

**Comment by 007durgesh219** [comment]
OK, In that case using assertEqualXMLStructure seems a better option to me [https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.assertions.assertEqualXMLStructure](url)  Also, domDocument provide method to validate HTML [http://php.net/manual/en/domdocument.validate.php](url)

**Comment by zixtor** [comment]
That seems good and sufficient to me as well, please feel free to proceed.

**Comment by 007durgesh219** [comment]
Analyzing composer.json/composer.lock files to determine dependencies of the root package. 
I didn't get why it failed here. Kindly help

**Comment by 007durgesh219** [comment]
@zixtor Kindly look into this issue and help me figure out what I did wrong here

**Comment by zixtor** [comment]
I think Scrutinizer error is not due to your changes, and the tests seem to be passing now.

File: templates/getCheckbox.phtml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        <?php }
+        if (isset($checked) && $checked) {      ?>
+            checked="checked"
+        <?php }
+        if (isset($onclick) && $onclick) {      ?>
+            class="autosubmit"
+        <?php } ?> />
+<label  <?php if (isset($html_field_id)) {      ?>
+            for="<?php echo $html_field_id ; }  ?>">
+        <?php echo $label ; ?> </label>
```
Comment (zixtor) [inline templates/getCheckbox.phtml]: This file should be named 'checkbox.phtml'

**Comment by 007durgesh219** [comment]
OK, Thanks, Please review and let me know if any changes required. I've changed test also, Now we don't have to maintain test if  the relative position of attributes changed.

File: templates/getWithSelected.phtml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<img class="selectallarrow" src="<?php echo $pmaThemeImage . 'arrow_' . $text_dir . '.png' ; ?>"
+    width="38" height="22" alt="<?php echo __('With selected:') ; ?>" />
+<input type="checkbox" id="<?php echo $formName . '_checkall' ;   ?>" class="checkall_box"
+    title="<?php echo __('Check all') ;      ?>" />
+<label for="<?php echo $formName . '_checkall' . __('Check all') ;?>" </label>
+<i style="margin-left: 2em">
+    <?php echo __('With selected:') ;                             ?> </i>
```
Comment (zixtor) [inline templates/getWithSelected.phtml]: For this, maybe 'select_all.phtml'

File: templates/getCheckbox.phtml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<input type="checkbox" name="<?php echo $html_field_name; ?>"
+        <?php if (isset($html_field_id)) {      ?>
+            id="<?php echo $html_field_id ;     ?>"
```
Comment (zixtor) [inline templates/getCheckbox.phtml]: use <?= in all places where you would do <?php echo

File: templates/getCheckbox.phtml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<input type="checkbox" name="<?php echo $html_field_name; ?>"
+        <?php if (isset($html_field_id)) {      ?>
```
Comment (zixtor) [inline templates/getCheckbox.phtml]: avoid curly braces, see http://php.net/manual/en/control-structures.alternative-syntax.php

File: test/libraries/common/PMA_getCheckbox_test.php [hunk 5]
```diff
@@ -30,11 +30,13 @@ function testGetCheckbox()
...
-            PMA\libraries\Util::getCheckbox($name, $label, false, false, $name),
-            '<input type="checkbox" name="' . $name . '" id="' . $name
+        // assertXmlStringEqualsXmlString require both inputs to be a valid xml string
+        // dummy <root> tag will make input a valid xml string
+        $this->assertXmlStringEqualsXmlString(
+            '<root>' . PMA\libraries\Util::getCheckbox($name, $label, false, false, $name) . '</root>',
+            '<root> <input type="checkbox" name="' . $name . '" id="' . $name
             . '" /><label for="' . $name . '">' . $label
-            . '</label>'
+            . '</label> </root>'
```
Comment (zixtor) [inline test/libraries/common/PMA_getCheckbox_test.php]: This seems better, thanks!

**Comment by zixtor** [comment]
As an additional improvement, the functions get getWithSelected() and getCheckbox() can be removed now, and in all places where they are used, the extracted templates can be called instead.

**Comment by 007durgesh219** [comment]
OK, will do all those changes and update PR

**Comment by 007durgesh219** [comment]
Also, I think it will be more convenient to change test file names, Like test/libraries/common/PMA_getCheckbox_test.php -> test/libraries/common/checkbox_test.php and for other one also

**Comment by zixtor** [comment]
Yes, makes sense.

**Comment by 007durgesh219** [comment]
@zixtor I've done all changes you suggested. Please review PR

**Comment by zixtor** [comment]
Thanks @007durgesh219 for the contribution!

**Pull request merged**